File: explore/hash_table/contains_duplicate.py
# Given an array of integers, find if the array contains any duplicates.
# The easy way to do this is by converting the array into a set which doesn't allow duplicates and then comparing the length. Or you could do it formally and populate the set.
# Alternatively, you could make a count hash map or a boolean hash map, but this is not necessary.
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hash_set = set(nums)

        return len(hash_set) != len(nums)

File: golang/contains_duplicate.go
package main

// given an integer array, return true if any value appears at least twice, false otherwise.
// create a hash set to check for duplicates.
func containsDuplicate(nums []int) bool {
	set := map[int]bool{}

	for _, num := range nums {
		if set[num] {
			return true
		}
		set[num] = true
	}

	return false
}
